pkg/controller/scyllacluster/sync.go

```diff
@@ -44,304 +47,193 @@ func (scc *Controller) sync(ctx context.Context, key string) error {
 		return err
 	}
 
+	type CT = *scyllav1.ScyllaCluster
+	var releaseErrs []error
+
 	scSelector := labels.SelectorFromSet(labels.Set{
 		naming.ClusterNameLabel: sc.Name,
 	})
 
-	type CT = *scyllav1.ScyllaCluster
-	var objectErrs []error
-
-	statefulSetMap, err := controllerhelpers.GetObjects[CT, *appsv1.StatefulSet](
+	err = controllerhelpers.ReleaseObjects[CT, *appsv1.StatefulSet](
```

Comment:
I wonder, if you were to create ScyllaDBDatacenter first and than release the objects - would that avoid the case where the objects can be left orphaned around or are there issues with that approach? The Three rules of controllers should prevent any unwanted action with overlapping selectors.

Reply:
I guess it would be the same, PR version is cleaner in my opinion.